gen/edu/uc/emiller/carapp/service/CarService.kt

```diff
@@ -0,0 +1,35 @@
+package edu.uc.emiller.carapp.service
+
+import android.app.Application
+import android.content.ContentValues.TAG
+import android.util.Log
+import androidx.room.Room
+import edu.uc.emiller.carapp.dao.CarDatabase
+import edu.uc.emiller.carapp.dao.ICarDAO
+import edu.uc.emiller.carapp.dto.Car
+
+class CarService(application: Application) {
+    private val application = application
+
+    /**
+     * Store cars locally using Room database
+     */
+    private suspend fun updateCars(cars: ArrayList<Car>?) {
+        try {
+            var carDAO = getCarDAO()
```

Comment:
Wrap this with a null check first.  That's easy to do with scope functions:

cars?.let{
carDAO.insertAll(cars)
}

The curlies will only execute if cars is not null.